cli/src/main/kotlin/com/malinskiy/marathon/cli/args/environment/SystemEnvironmentReader.kt

```diff
@@ -3,6 +3,13 @@ package com.malinskiy.marathon.cli.args.environment
 import com.malinskiy.marathon.cli.args.EnvironmentConfiguration
 import java.io.File
 
-class SystemEnvironmentReader : EnvironmentReader {
-    override fun read() = EnvironmentConfiguration(System.getenv("ANDROID_HOME")?.let { File(it) })
-}
\ No newline at end of file
+class SystemEnvironmentReader(
+    private val environment: (String) -> String?
```

Comment:
Should this have a default value of `{ it -> System.getenv(it)}`?

Reply:
You may decide.
My experience: Default values are nice, for example, in data classes. However, in logic classes they may make code and tests harder to understand.

Reply:
Since this is a SystemEnvironmentReader and I'd prefer to have a default value for the lambda here because it will encapsulate at least what the value for the lambda is expected to be

Reply:
Done